File: tests/wrapper_js/print_info.js
const fs = require('fs')
const path = require("path")

let INSTALL = undefined
let DATA_DIR = undefined

function main() {
    try
    {
        const ffmpegcpp = require(INSTALL+"FfmpegCpp.node")

        // Load a video from a container and read its info.
        const fileName = DATA_DIR + "/big_buck_bunny.mp4"

        const demuxer = new ffmpegcpp.Demuxer(fileName);
        const info = demuxer.GetInfo();

        console.log("Input " + info.format.name + " from '" + fileName + "'")

        console.log("Video streams: ")
        for (let i = 0; i < info.videoStreams.size(); ++i)
        {
            const stream = info.videoStreams.get(i)
            console.log("Stream #" + (i + 1) + ": codec " + stream.codec.name
                    + ", pixel format " + stream.formatName + ", resolution "
                    + stream.width + "x" + stream.height + ", bit rate "
                    + stream.bitRate + "kb/s"
                    + ", fps "
                    + stream.frameRate.num / stream.frameRate.den
                    + ", time base " + stream.timeBase.num + "/"
                    + stream.timeBase.den + ", " + demuxer.GetFrameCount(stream.id)
                    + " frames"
            )
        }

        console.log("Audio streams: ")
        for (let i = 0; i < info.audioStreams.size(); ++i) {
            const stream = info.audioStreams.get(i);
            console.log("Stream #" + (i + 1) + ": codec " + stream.codec.name
                    + ", channel layout " + stream.channelLayoutName + ", channels "
                    + stream.channels + ", bit rate " + stream.bitRate + "kb/s"
                    + ", sample rate " + stream.sampleRate+ ", time base "
                    + stream.timeBase.num + "/" + stream.timeBase.den + ", "
                    + demuxer.GetFrameCount(stream.id) + " frames")
        }
    }
    catch (err)
    {
        console.error("Failed to run print info " + err)
        return 1
    }

    return 0
}

if (__filename.indexOf(process.argv[1]) !== -1) {
    const args = process.argv.slice(2);

    for (let i = 0; i < args.length; i++)
    {
        if (args[i] === "--install")
        {
            if (fs.existsSync(args[i+1]))
            {
                INSTALL = path.resolve(args[i+1]) + "/nodejs/FfmpegCpp/"
            }
        }
        if (args[i] === "--data")
        {
            if (fs.existsSync(args[i+1]))
            {
                DATA_DIR = path.resolve(args[i+1])
            }
        }
    }

    process.exit(main())
}

File: tests/wrapper_python/print_info.py
# -*- coding: utf-8 -*-
# Python3 support only
#

import os
import sys
import argparse


def main() -> int:

    try:
        import FfmpegCpp.FfmpegCpp as ffmpegcpp
    except ModuleNotFoundError:
        print("[ERROR] FfmpegCpp Not Found.")
        return 1

    # Load a video from a container and read its info.
    fileName = DATA_DIR + "/big_buck_bunny.mp4"

    demuxer = ffmpegcpp.Demuxer(fileName)
    info = demuxer.GetInfo()

    print("Input " + info.format.name + " from '" + fileName + "'")
    print("Video streams: ")
    i = 0
    for stream in info.videoStreams:
        out = "Stream #%d" % (i + 1)
        out += ": codec " + stream.codec.name
        out += ", pixel format " + stream.formatName
        out += ", resolution %dx%d" % (stream.width, stream.height)
        out += ", bit rate %fkb/s" % (stream.bitRate)
        out += ", fps %f" % (stream.frameRate.num / stream.frameRate.den)
        out += ", time base %d/%d" % (stream.timeBase.num, stream.timeBase.den)
        out += ", %d frames" % (demuxer.GetFrameCount(stream.id))
        print(out)
        i = i + 1

    print("Audio streams: ")
    i = 0
    for stream in info.audioStreams:
        out = "Stream #%d" % (i + 1)
        out += ": codec " + stream.codec.name
        out += ", channel layout " + stream.channelLayoutName
        out += ", channels %d" % (stream.channels)
        out += ", bit rate %fkb/s" % (stream.bitRate)
        out += ", sample rate %d" % (stream.sampleRate)
        out += ", time base %d/%d" % (stream.timeBase.num, stream.timeBase.den)
        out += ", %d frames" % (demuxer.GetFrameCount(stream.id))
        print(out)
        i = i + 1
    return 0


if __name__ == "__main__":
    try:
        install_index = sys.argv.index("--install") + 1
        install = sys.argv[install_index]
        data_index = sys.argv.index("--data") + 1
        DATA_DIR = sys.argv[data_index]
    except ValueError:
        pass

    sys.path.insert(0, os.path.abspath(os.path.join(install, 'python')))
    sys.exit(main())
